packages/common/dynamoFunctions/tests/stateMapping.test.ts

```diff
@@ -0,0 +1,182 @@
+import {jest} from "@jest/globals"
+
+import {
+  DynamoDBDocumentClient,
+  PutCommand,
+  GetCommand,
+  DeleteCommand
+} from "@aws-sdk/lib-dynamodb"
+import {Logger} from "@aws-lambda-powertools/logger"
+import {insertStateMapping, getStateMapping, deleteStateMapping} from "../src/stateMapping"
+
+const mockLogger: Partial<Logger> = {
+  info: jest.fn(),
+  debug: jest.fn(),
+  error: jest.fn()
+}
+
+describe("insert stateMapping", () => {
+  let mockDocumentClient: jest.Mocked<DynamoDBDocumentClient>
+
+  beforeEach(() => {
+    jest.clearAllMocks()
+    mockDocumentClient = {
+      send: jest.fn()
+    } as unknown as jest.Mocked<DynamoDBDocumentClient>
+  })
+
+  it("should insert into DynamoDB successfully", async () => {
+    // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
+    mockDocumentClient.send.mockResolvedValueOnce({} as never)
```

Comment:
This can be done without the `eslint` ignore statement with something like this:
```typescript
mockDocumentClient.send.mockImplementationOnce(() => Promise.reject(new Error()))
```
The existing version works fine and this doesn't block me approving the PR, I just have a preference for avoiding overwriting `eslint` rules